network/p2p/p2p.go

```diff
@@ -80,7 +82,7 @@ type p2pNetwork struct {
 }
 
 // New creates a new p2p network
-func New(logger *zap.Logger, cfg *Config) network.P2PNetwork {
+func New(logger *zap.Logger, cfg *Config, mr *metricsreporter.MetricsReporter) network.P2PNetwork {
```

Comment:
instead of requiring a whole `MetricsReporter` implementation, i suggest to require instead a package-level `Metrics` interface that includes only the methods that `p2pNetwork` needs to call

that's to be consistent with other packages (such as `protocol/v2/ssv/queue`)